### PR TITLE
Replace TrainValidationSplit with CrossValidation

### PR DESCRIPTION
**Pull request by amCap1712**
CrossValidation splits the training data multiple times as opposed to TrainValidationSplit which only does it once. Therefore, CrossValidation takes more time to train but can also help in less randomness in models.

### CODE REVIEW

File: listenbrainz_spark/recommendations/recording/tests/test_models.py [hunk 1]
```diff
@@ -73,8 +72,9 @@ def test_get_best_model(self, mock_als_cls, mock_params, mock_tvs):
             estimator=mock_als_cls.return_value,
             estimatorParamMaps=mock.ANY,
             evaluator=mock_evaluator,
-            trainRatio=0.80,
-            collectSubModels=True
+            numFolds=5,
```
Comment (mayhem) [inline listenbrainz_spark/recommendations/recording/tests/test_models.py:75]: Can we please document these values? And maybe make constants for them, if we anticipate changing them?

File: listenbrainz_spark/recommendations/recording/train_models.py [hunk 2]
```diff
@@ -148,20 +155,21 @@ def train_models(training_data, evaluator, ranks, lambdas, iterations, alphas, c
...
         .build()
 
-    tvs = TrainValidationSplit(
+    cv = CrossValidator(
         estimator=als,
         estimatorParamMaps=params,
         evaluator=evaluator,
-        trainRatio=0.80,
-        collectSubModels=True
+        numFolds=5,
```
Comment (mayhem) [inline listenbrainz_spark/recommendations/recording/train_models.py:162]: Yes, lets make constants if they are being used in two places.

**Review by mayhem — APPROVED** [review]
One minor change, then good to go.

File: listenbrainz_spark/recommendations/recording/tests/test_models.py (hunk 1 above)
Comment (amCap1712) [reply listenbrainz_spark/recommendations/recording/tests/test_models.py]: Done

**Comment by mayhem** [comment]
:+1:

**Pull request merged**